lib/alerts.ex

```diff
@@ -37,6 +37,30 @@ defmodule Alerts do
   def by_service(alerts, route_type_string),
     do: Enum.filter(alerts, &Alert.matches_service_type(&1, String.to_integer(route_type_string)))
 
+  @doc """
+  Return a map with route ids as string keys and a list of alerts as the value
+  """
+  @spec by_route([Alert.t()]) :: map()
+  def by_route(bus_alerts) do
+    Enum.reduce(bus_alerts, %{}, fn alert, acc ->
+      Enum.map(alert.informed_entity, & &1.route)
+      |> Enum.reduce(%{}, fn route_id, route_ids_acc ->
+        Map.merge(route_ids_acc, %{
+          route_id => [
+            %{
+              severity: alert.severity,
+              created_at: alert.created_at,
+              id: alert.id
+            }
+          ]
+        })
+      end)
+      |> Map.merge(acc, fn _k, route_ids_map, acc ->
```

Comment:
thought: So much nested data! It definitely took me a bit to figure out what's going on here (but I also don't know that I have anything better to suggest).

Reply:
I think I can rewrite this one to be a bit clearer, got a little carried away here maybe.